File: scalechain-wallet/src/main/scala/io/scalechain/wallet/WalletSampleData.scala
package io.scalechain.wallet

import io.scalechain.blockchain.chain.{ChainSampleData, ChainEventListener}
import io.scalechain.blockchain.storage.index.KeyValueDatabase
import io.scalechain.blockchain.transaction.{CoinAddress, TransactionTestDataTrait}


class WalletSampleData(wallet:Wallet)(protected override val db : KeyValueDatabase) extends ChainSampleData(Some(wallet))(db) {
  override def onAddressGeneration(account: String, address: CoinAddress): Unit = {
    wallet.importOutputOwnership(TestBlockchainView, account, address, false)(db)
  }
}

File: scalechain-wallet/src/main/kotlin/io/scalechain/wallet/WalletSampleData.kt
package io.scalechain.wallet

import io.scalechain.blockchain.chain.ChainSampleData
import io.scalechain.blockchain.chain.ChainEventListener
import io.scalechain.blockchain.storage.index.KeyValueDatabase
import io.scalechain.blockchain.transaction.CoinAddress


open class WalletSampleData(db : KeyValueDatabase, private val wallet : Wallet) : ChainSampleData(db, wallet) {
  override fun onAddressGeneration(account: String, address: CoinAddress): Unit {
    // BUGBUG : Kotlin bug : wallet is null.
    // assert(wallet != null)
    // wallet.importOutputOwnership(db, TestBlockchainView, account, address, false)

    val w = chainEventListener as Wallet
    w.importOutputOwnership(db, TestBlockchainView, account, address, false)
  }
}
